Gems/Atom/Feature/Common/Code/Source/DiffuseGlobalIllumination/DiffuseProbeGridRelocationPass.cpp

```diff
@@ -74,49 +74,71 @@ namespace AZ
             if (!diffuseProbeGridFeatureProcessor || diffuseProbeGridFeatureProcessor->GetVisibleRealTimeProbeGrids().empty())
             {
                 // no diffuse probe grids
-                return;
+                return false;
             }
 
-            RayTracingFeatureProcessor* rayTracingFeatureProcessor = scene->GetFeatureProcessor<RayTracingFeatureProcessor>();
-            AZ_Assert(rayTracingFeatureProcessor, "DiffuseProbeGridRelocationPass requires the RayTracingFeatureProcessor");
-
-            if (!rayTracingFeatureProcessor->GetSubMeshCount())
+            // check TLAS version
+            uint32_t rayTracingDataRevision = rayTracingFeatureProcessor->GetRevision();
+            if (rayTracingDataRevision != m_rayTracingDataRevision)
             {
-                // empty scene
-                return;
+                return true;
             }
 
-            // create the Relocation Srgs for each DiffuseProbeGrid, and check to see if any grids need relocation
-            bool needRelocation = false;
+            // check to see if any grids need relocation           
             for (auto& diffuseProbeGrid : diffuseProbeGridFeatureProcessor->GetVisibleRealTimeProbeGrids())
             {
-                uint32_t rayTracingDataRevision = rayTracingFeatureProcessor->GetRevision();
-                if (rayTracingDataRevision != m_rayTracingDataRevision)
+                if (diffuseProbeGrid->GetRemainingRelocationIterations() > 0)
                 {
-                    // the TLAS changed, relocate probes
-                    m_rayTracingDataRevision = rayTracingDataRevision;
-                    diffuseProbeGrid->ResetRemainingRelocationIterations();
+                    return true;
                 }
+            }
+           
+            return false;
+        }
 
-                if (diffuseProbeGrid->GetRemainingRelocationIterations() > 0)
+        void DiffuseProbeGridRelocationPass::FrameBeginInternal(FramePrepareParams params)
+        {
+            RPI::Scene* scene = m_pipeline->GetScene();
+            DiffuseProbeGridFeatureProcessor* diffuseProbeGridFeatureProcessor = scene->GetFeatureProcessor<DiffuseProbeGridFeatureProcessor>();
+            RayTracingFeatureProcessor* rayTracingFeatureProcessor = scene->GetFeatureProcessor<RayTracingFeatureProcessor>();
+            AZ_Assert(rayTracingFeatureProcessor, "DiffuseProbeGridRelocationPass requires the RayTracingFeatureProcessor");
+
+            // reset the relocation iterations on the grids if the TLAS was updated
+            uint32_t rayTracingDataRevision = rayTracingFeatureProcessor->GetRevision();    
+            for (auto& diffuseProbeGrid : diffuseProbeGridFeatureProcessor->GetVisibleRealTimeProbeGrids())
+            {                
+                if (rayTracingDataRevision != m_rayTracingDataRevision)
```

Comment:
Very minor, but this if could be around the for loop instead of inside it, no?

Reply:
Yes, I'll move this out.